spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/RegionRroposal.scala

```diff
@@ -0,0 +1,342 @@
+  }
+
+  override def accGradParameters(input: Table, gradOutput: Tensor[Float]): Unit = {
+    throw new UnsupportedOperationException("RegionRroposal only support inference")
+  }
+
+  override def parameters(): (Array[Tensor[Float]], Array[Tensor[Float]]) = {
+    head.parameters()
+  }
+
+  override def getParametersTable(): Table = {
+    head.getParametersTable()
+  }
+
+  override def clearState(): this.type = {
+    super.clearState()
+    head.clearState()
+    boxSelector.clearState()
+    this
+  }
+
+  override def release(): Unit = {
+    super.release()
+    head.release()
+    boxSelector.release()
+  }
+
+  override def training(): RegionRroposal.this.type = {
+    train = true
+    head.training()
+    boxSelector.training()
+    super.training()
+  }
+
+  override def evaluate(): this.type = {
+    head.evaluate()
+    boxSelector.evaluate()
+    train = false
+    super.evaluate()
+  }
+}
+
+object RegionRroposal {
+  def apply(inChannels: Int,
+            anchorSizes: Array[Float] = Array[Float](32, 64, 128, 256, 512),
+            aspectRatios: Array[Float] = Array[Float](0.5f, 1.0f, 2.0f),
+            anchorStride: Array[Float] = Array[Float](4, 8, 16, 32, 64),
+            preNmsTopNTest: Int = 1000,
+            postNmsTopNTest: Int = 1000,
+            preNmsTopNTrain: Int = 2000,
+            postNmsTopNTrain: Int = 2000,
+            nmsThread: Float = 0.7f,
+            minSize: Int = 0)(implicit ev: TensorNumeric[Float]): RegionRroposal =
+    new RegionRroposal(inChannels, anchorSizes, aspectRatios, anchorStride,
+      preNmsTopNTest, postNmsTopNTest, preNmsTopNTrain, postNmsTopNTrain, nmsThread,
+      minSize)
+}
+
+private[nn] class ProposalPostProcessor(
+  val preNmsTopNTest: Int = 1000,
+  val postNmsTopNTest: Int = 1000,
+  val preNmsTopNTrain: Int = 2000,
+  val postNmsTopNTrain: Int = 2000,
+  val nmsThread: Float = 0.7f,
+  val minSize: Int = 0)
+  (implicit ev: TensorNumeric[Float]) extends AbstractModule[Table, Table, Float]{
+
+  @transient private var sortedScores: Tensor[Float] = null
+  @transient private var sortedInds: Tensor[Float] = null
+  @transient private var boxRegressionIndex: Tensor[Float] = null
+  @transient private var anchorsIndex: Tensor[Float] = null
+
+  private val nms = new Nms()
+  private val arr = new Array[Int](10000)
+  private val sigmoid = Sigmoid[Float]()
+
+  /**
+   * Arguments:
+   *    anchors: Tensor with shape (N, nums, 4)
+   *    objectness: Tensor of size N, A, H, W
+   *    box_regression: Tensor of size N, A * 4, H, W
+   *    img_info: image size
+   * @param input
+   * @return
+   */
+  override def updateOutput(input: Table): Table = {
+    //  for memory case, input may be changed
+    val anchors = input[Tensor[Float]](1)
+    var objectness = input[Tensor[Float]](2)
+    var boxRegression = input[Tensor[Float]](3)
+    val imageSize = input[Tensor[Float]](4) // height, width
+
+    val N = objectness.size(1)
```

Comment:
what does N, A, H, W mean ? can we name them more meaningful ?

Reply:
add some comments for those varaibles